### PR TITLE
AMS-3313 added To field into mail importer bean.

### PR DESCRIPTION
**Pull request by leighfwarren**

### CODE REVIEW

**Review by themnd — COMMENTED** [review]
why do you need the "to" field?
Each route can import mails from a specifc route, so the "To" field seems unnecessary.

**Comment by leighfwarren** [comment]
> why do you need the "to" field?
> Each route can import mails from a specifc route, so the "To" field seems unnecessary.

It's is to support different destinations from the same mailbox.  For example, if we wanted to send the article to a review partition we could send the email to:
dmdesk+incoming@teckbote.de
We can then retrieve the To address and then pick it apart.  This means that we only need to have a single mailbox.

File: src/main/java/com/atex/plugins/mailimporter/MailParserImpl.java [hunk 1]
```diff
@@ -49,6 +52,9 @@ public MailBean parse(final Exchange exchange) throws Exception {
         MailBean mailBean = createMailBean();
 
         String subject = messageParser.getSubject();
+        List<Address> to = messageParser.getTo();
+        String toAddress = to.stream().map(address -> address.toString()).collect(Collectors.joining(","));
```
Comment (themnd) [inline src/main/java/com/atex/plugins/mailimporter/MailParserImpl.java:56]: are you sure the to list is always non null?

Comment (leighfwarren) [reply src/main/java/com/atex/plugins/mailimporter/MailParserImpl.java:56]: MimeMessageParser always returns a list:

    public List<Address> getTo() throws Exception {
        Address[] recipients = this.mimeMessage.getRecipients(RecipientType.TO);
        return (List)(recipients != null ? Arrays.asList(recipients) : new ArrayList());
    }

**Review by themnd — APPROVED** [review]

**Pull request merged**